### PR TITLE
Update Kubernetes client libraries to 1.18

### PR DESCRIPTION
**Pull request by howardjohn**
This change upgrades our libraries. There should be no behavioral changes.

* client-go added a bunch of `contexts` everywhere. This was updated with https://github.com/kubernetes-sigs/clientgofix
* An empty kubeconfig is now an error. Some tests were relying on this, requiring some tiny tweaks to get this moving
* protobuf library changed, requiring golden file updates to conformance test. This is not a breaking change, since we don't depend on json format

### CODE REVIEW

**Comment by howardjohn** [comment]
It works! need to wait for k8s to publish everything though

**Comment by howardjohn** [comment]
Fixes https://github.com/istio/istio/issues/22081

**Comment by howardjohn** [comment]
> Waiting on code owner review from costinm, linsun, rshriram, istio/wg-environments-maintainers, istio/wg-networking-maintainers, istio/wg-networking-maintainers-pilot, istio/wg-policies-and-telemetry-maintainers, istio/wg-security-maintainers, istio/wg-test-and-release-maintainers, and/or istio/wg-user-experience-maintainers. 

Well this should be fun

**Review by rshriram — APPROVED** [review]

File: operator/pkg/manifest/client.go [hunk 1]
```diff
@@ -151,7 +152,7 @@ func (client *Client) GetPods(namespace string, params map[string]string) (*v1.P
 		req.Param(k, v)
 	}
 
-	res := req.Do()
+	res := req.Do(context.TODO())
```
Comment (mandarjog) [inline operator/pkg/manifest/client.go]: initialization should use `context.Background()`  `TODO` is if we don't know or cannot send ctx yet.

Here and elsewhere. I see that TODO is used everywhere.

File: operator/cmd/mesh/operator_test.go [hunk 2]
```diff
@@ -163,14 +166,14 @@ func TestOperatorRemove(t *testing.T) {
 				operatorNamespace: "operator-test-namespace",
 				istioNamespace:    "istio-test-namespace",
 			},
+			kubeConfigPath: path.Join(env.IstioSrc, "tests/util/kubeconfig"),
```
Comment (mandarjog) [inline operator/cmd/mesh/operator_test.go]: Is this a related change?

File: pilot/pkg/serviceregistry/kube/controller/controller.go [hunk 3]
```diff
@@ -579,6 +580,28 @@ func (c *Controller) GetProxyServiceInstances(proxy *model.Proxy) ([]*model.Serv
 	return out, nil
 }
 
+func getPodServices(s listerv1.ServiceLister, pod *v1.Pod) ([]*v1.Service, error) {
```
Comment (mandarjog) [inline pilot/pkg/serviceregistry/kube/controller/controller.go]: Is this related change or refactoring? hard to tell.

File: operator/pkg/manifest/client.go (hunk 1 above)
Comment (hzxuzhonghu) [reply operator/pkg/manifest/client.go]: IMO, TODO is more suitable, Background is more often used to initiate a complete context.

**Review by liamawhite — APPROVED** [review]

**Review by nmittler — APPROVED** [review]

**Review by ericvn — APPROVED** [review]

**Review by myidpt — APPROVED** [review]

**Review by elfinhe — APPROVED** [review]

**Review by mandarjog — APPROVED** [review]
/lgtm
